### ISSUE TITLE
Cannot build qt5 due to fonts

### ISSUE THREAD

**Issue by olesenm** [spack/spack#6685]
Couldn't build qt-5.9.1 on my system and figured I'd try it with new versions:

     version('5.10.0', 'c5e275ab0ed7ee61d0f4b82cd471770d')
     version('5.9.3',  'eae2435509493a2084405714e0a9bdf9')

No better. Dug through the build output and it was first choking on `-system-freetype` and after hacking around that, it was choking on `-fontconfig`. It's still chugging away (I have lots building today), but these hacks seem to have helped:

    @@ -209,8 +210,8 @@ class Qt(Package):
                 '-confirm-license',
                 '-openssl-linked',
                 '-optimized-qmake',
    -            '-fontconfig',
    -            '-system-freetype',
    +            ## NO '-fontconfig',
    +            ## NO '-system-freetype',
                 '-I{0}/freetype2'.format(self.spec['freetype'].prefix.include),
                 '-no-pch'
             ]

Not the most conclusive issue report, but thought I'd raise it anyhow. It looks like @adamjstewart or @davydden are the ones who might know what's going on here.

Cheers,
/mark

**Comment by adamjstewart** [comment]
Unfortunately, the only thing I know about `qt` is that it's one of the hardest packages to get installed. I have my own slew of issues with it: #4959, #4960, #4961.

I do know that we're missing about half of `qt`'s dependencies, so adding those may help. But it's too complicated of a package for me to want to slog through. I've given up on it for the time being. If anyone else wants to be a hero, this package would be a great place to start.

**Comment by olesenm** [comment]
I understand. For now I can just drag along my commented-out hack. The only reason I need qt is for paraview, but that adds even more possibilities that we haven't even really addressed (opengl vs mesa, or opengl for paraview, but not for qt, etc).

**Comment by davydden** [comment]
I have the same issue with fonts
```
ERROR: Feature 'fontconfig' was enabled, but the pre-condition '!config.win32 && !config.darwin && features.system-freetype && libs.fontconfig' failed.
```

**Comment by davydden** [comment]
@olesenm @adamjstewart `fontconfig` is `(Unix only)` and should be disabled on macOS. I am preparing a PR (it's actually ready, just waiting for the build to hopefully succeed).

**Comment by olesenm** [comment]
This is good news. Interestingly though, my problem showed up on opensuse 42.3 (ie, Linux)

**Comment by davydden** [comment]
Hm, that’s strange... then my PR won’t fix this issue

**Comment by alalazo** [comment]
Closing as I was able to build `qt@5.13.1` on `linux` at 3cd6ca02d2d58c42d6864b25ed5e4479557eaa8c and the issue for Mac OS is apparently solved by #6766. Feel free to reopen if the issue persists for you.
